Source/Models/JSONDeserializer.swift

```diff
@@ -0,0 +1,37 @@
+import Argo
+import Result
+
+struct JSONDeserializer: Deserializer {
+  func deserialize<T : Request>(request: T) -> (NSData?) -> Result<T.ResponseParser.Representation, SwishError> {
```

Comment:
Yeah, its just so `T.ResponseParser` is concrete. I don't like it either, but moving the deserializer to the `performRequest` level makes the API feel way more complicated, and kind of confuses the point of this PR, because now instead of having different clients for different response parsing, you can have one client that can decode any number of different response types.